machine-learning/Luna-Forecasting.ipynb

```diff
@@ -0,0 +1,507 @@
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "// call df.Head(10) if you want to see the first 10 line.\n",
+        "df.Tail(10)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Construct training pipeline\n",
+        "The following code shows how to construct pipeline, which simply concatenate all columns starts with \"prev\" as a single-vector feature and feed it into regressors.\n",
+        "\n",
+        "Notice that the pipeline calls `mlContext.Auto().Regression` in the end, which returns all available regressors with pre-defined search space. Then we can pass this pipeline into `AutoMLExperiemnt` for hpo.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "var rowCount = df.Rows.Count();\n",
```

Comment:
I'd add more details on what you're doing here. Maybe even splitting up the code into the individual steps.